react/ProductContextProvider.tsx

```diff
@@ -110,8 +99,14 @@ const ProductContextProvider: FC<ProductAndQuery> = ({
 }) => {
   const [state, dispatch] = useProductReducer({ query, product })
 
-  // These hooks are used to keep the state in sync with API data, specially when switching between products without exiting the product page
-  useProductInState(product, dispatch)
+  // It is used to keep the state in sync with API data, specially when switching between products without exiting the product page
+  if (product && (product.productId !== state?.product?.productId)) {
```

Comment:
👀

Reply:
I'm wondering if other attribute that is not the id, change, what will occur?

Reply:
Nope

I think @kaisermann is pointing to the lint issue, maybe?

Reply:
what breno said 🙏